test/evergreen/evg_cfg.py

```diff
@@ -0,0 +1,335 @@
+#! /usr/bin/env python3
+
+"""
+This program provides an CLI interface to check and generate Evergreen configuration.
+"""
+
+import os
+import sys
+import re
+import subprocess
+import docopt
+
+TEST_TYPES = ('make_check', 'csuite')
+EVG_CFG_FILE = "test/evergreen.yml"
+MAKE_SUBDIRS_FILE = "build_posix/Make.subdirs"
+CSUITE_TEST_DIR = "test/csuite"
+MAKE_CHECK_TEST_TMPLT = "test/evergreen/make_check_test_evg_task.template"
+CSUITE_TEST_TMPLT = "test/evergreen/csuite_test_evg_task.template"
+MAKE_CHECK_TEST_SEARCH_STR = "  # End of normal make check test tasks"
+CSUITE_TEST_SEARCH_STR = "  # End of csuite test tasks"
+WIREDTIGER_REPO = "git@github.com:wiredtiger/wiredtiger.git"
+
+# This list of sub directories will be skipped from checking
+# They are not expected to trigger any 'make check' testing.
+make_check_subdir_skips = [
+    "ext/collators/reverse",
+    "ext/collators/revint",
+    "ext/compressors/lz4",
+    "ext/compressors/nop",
+    "ext/compressors/snappy",
+    "ext/compressors/zlib",
+    "ext/compressors/zstd",
+    "ext/datasources/helium",
+    "ext/encryptors/nop",
+    "ext/encryptors/rotn",
+    "ext/extractors/csv",
+    "ext/test/kvs_bdb",
+    "ext/test/fail_fs",
+    "api/leveldb",
+    "lang/java",
+    "test/utility",
+    "examples/java",
+    "test/csuite",  # csuite has its own set of Evergreen tasks, skip the checking here
+    "test/syscall",
+    "bench/workgen",
```

Comment:
How was this list generated? I'm confused, because I doubt `lang/python` has `make check` testing, either? My guess is there are more directories that don't involve `make check` testing, than do involve it.

That said, could we instead check the `Makefile` in a directory, and see if there's a `TESTS =` target?

Or, have an empty set of tests in directories that don't include testing so we don't have to skip directories?

Reply:
Here is the set of directories I got while grep `TESTS =` from Makefile.am as below. `api/leveldb` showed up here, but was not in the `make check` testing scope for some reason. All other directories make sense (`test/csuite` is split up separately and will be an exception we need to skip). I'll go ahead implementing this solution, and likely need to keep the skip list (with limited number of entries though) to exclude `api/leveldb` and `test/csuite`.

> $ find . -name Makefile.am -exec grep -H "^TESTS =" {} \;
> ./api/leveldb/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./bench/wtperf/Makefile.am:TESTS = smoke.sh
> ./examples/c/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./lang/python/Makefile.am:TESTS = run-ex_access
> ./test/bloom/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./test/checkpoint/Makefile.am:TESTS = smoke.sh
> ./test/csuite/Makefile.am:TESTS = $(all_TESTS)
> ./test/cursor_order/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./test/fops/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./test/format/Makefile.am:TESTS = smoke.sh
> ./test/huge/Makefile.am:TESTS = smoke.sh
> ./test/manydbs/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./test/packing/Makefile.am:TESTS = smoke.sh
> ./test/readonly/Makefile.am:TESTS = smoke.sh
> ./test/salvage/Makefile.am:TESTS = $(noinst_PROGRAMS)
> ./test/thread/Makefile.am:TESTS = smoke.sh